compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirMemberPropertyChecker.kt

```diff
@@ -5,32 +5,132 @@
+        //   constructor() { x = ... }
+        //   constructor(...): this() { ... }  // x will be initialized via this() delegation
+
+        // 3) Property can be initialized at any of class's anonymous initializers (all of initializers will be executed), e.g.,
+        //   init { x = ... }
+        //   ...
+        //   init { y = ... }
+
+        // 4) Property can be initialized at other property's initializing expression too, e.g.,
+        //   val initX = inlineMe { x = ... } // where inlineMe returns the value of the last expression of the lambda
+
+        // Here, we accumulate initialized info of a property per constructor, anonymous initializer, or property initialization
+        val initializedInConstructor = mutableMapOf<FirProperty, EventOccurrencesRange>().withDefault { EventOccurrencesRange.ZERO }
+        val initializedInInit = mutableMapOf<FirProperty, EventOccurrencesRange>().withDefault { EventOccurrencesRange.ZERO }
+        val initializedInOtherProperty = mutableMapOf<FirProperty, EventOccurrencesRange>().withDefault { EventOccurrencesRange.ZERO }
+
+        // To handle the delegated constructor call, we need a cache from constructor to (analyzed) property init info.
+        val constructorToData = mutableMapOf<FirConstructor, PropertyInitializationInfo>().withDefault { PropertyInitializationInfo.EMPTY }
+
+        fun collectInfoFromGraph(
+            graph: ControlFlowGraph,
+            map: MutableMap<FirProperty, EventOccurrencesRange>,
+            acc: (EventOccurrencesRange, EventOccurrencesRange) -> EventOccurrencesRange,
+            delegatedConstructor: FirConstructor? = null,
+        ) {
+            val delegatedInfo =
+                if (delegatedConstructor != null && constructorToData.containsKey(delegatedConstructor)) {
+                    constructorToData.getValue(delegatedConstructor)
+                } else PropertyInitializationInfo.EMPTY
+
+            val data = PropertyInitializationInfoCollector(memberProperties.map { it.symbol }.toSet()).getData(graph)
+            val infoAtExitNode = data[graph.exitNode]?.get(NormalPath) ?: PropertyInitializationInfo.EMPTY
+
+            val info = when {
+                delegatedInfo.isEmpty() -> infoAtExitNode
+                infoAtExitNode.isEmpty() -> delegatedInfo
+                else -> {
+                    // NB: it's not merge, which is conducted at merging points, such as loop condition or when conditions.
+                    // Rather, delegated constructor call is the predecessor of the current constructor call, so we should accumulate.
+                    delegatedInfo.plus(infoAtExitNode)
+                }
+            }
+
+            if (graph.declaration is FirConstructor) {
+                constructorToData.putIfAbsent(graph.declaration as FirConstructor, info)
             }
+
+            for (property in memberProperties) {
+                if (map.containsKey(property)) {
```

Comment:
How about simplify this whole branch with the following?
```
map[property] = acc.invoke(map.getValue(property), info[property.symbol] ?: EventOccurencesRange.ZERO)
```

Reply:
Tried; got regressions; and realized why I split the initial assignment case. Added code comments to explain that.

Reply:
Ah! I see, `ZERO` is not the identity of `or`.